### ISSUE TITLE
Does the endpoint become part of the cache key?

### ISSUE THREAD

**Issue by jkjustjoshing** [habx/apollo-multi-endpoint-link#155]
I have several GraphQL endpoints I'm trying to fetch from that share many of the same field names, but have different schemas and values within those fields. For example, what if I have these 2 queries I want to run:
```
query {
    products {
        id # UUID string
        seller # string
        price # Int, number of cents
    }
}
```
```
query {
    products {
        id # number
        affiliate # string
        price # string of the format "$1.23"
    }
}
```
Can I use this library to query from both of these endpoints without any caching issues?

**Comment by jean9696** [comment]
@jkjustjoshing No the endpoint name is not added in the cache key. 
But do your `products` grapqhl query return the same `__typename` ? Apollo is using it for caching so if the're not the same, it should work out of the box. Otherwise, you'll need to manage this yourself, our library does not handle caching. 

But this is an interesting question, I'll have a look and see if we can do something built-in

**Comment by jkjustjoshing** [comment]
@jean9696 the `__typename`s are the same - it's 2 different versions of the same API with different underlying data.

I have different SPA base routes for each API endpoint, so right now I'm throwing away the client/cache and creating a new one whenever switching between the SPA base routes. This should continue to work, but is a little clunky. And when going from section 1 to 2 then back to 1, you lose the cache.

**Comment by jean9696** [comment]
@jkjustjoshing What apollo cache are you using ?  Because if you're using `apollo-cache-inmemory` I do not understand why your cache would have issues because your ids do not overlap each others (uuid VS integer)

**Comment by jkjustjoshing** [comment]
@jean9696 the IDs we get back have a lot of overlap, unfortunately. No UUIDs separate for each instance. We're storing criminal justice data for different counties, one endpoint per county. So the number of people incarcerated is different in each county, but the ID we use to identify that the number is "number of people incarcerated" is always "23". So the object has a typename `datapoint` and a data type ID `23`, but nothing to distinguish it from county A and county B.
(this isn't *exactly* how the API is set up, but it's pretty close without getting into the intricacies of our data model)

But yeah, we're using `apollo-cache-inmemory`.

**Comment by jean9696** [comment]
@jkjustjoshing  I think we can handle it. I made a draft PR https://github.com/habx/apollo-multi-endpoint-link/pull/161 and will work on it to release a new minor soon 👍

**Comment by jean9696** [comment]
@jkjustjoshing released in [v2.5.0](https://www.npmjs.com/package/@habx/apollo-multi-endpoint-link/v/2.5.0), you juste need to add `prefixTypenames: true` in the constructor
